src/pages/stake/StakedDonut.tsx

```diff
@@ -64,24 +64,22 @@ const StakedDonut = ({ chain }: { chain?: string }) => {
 
   const RenderTooltip = (props: { payload?: any }) => {
     const { payload } = props
+    if (!payload?.length) return null
+    const { denom, amount, value, name, moniker } = payload[0].payload
 
     return (
       <div className={styles.tooltip}>
-        <h6>{payload[0]?.payload.name || payload[0]?.payload.moniker}</h6>
+        <h6>{name || moniker}</h6>
         <div className={styles.infoLine}>
           <p>Balance: </p>
           <p>
-            <Read
-              amount={payload[0]?.payload.amount}
-              fixed={2}
-              token={payload[0]?.payload.denom}
-            />
+            <Read amount={amount} fixed={2} denom={denom} />
           </p>
         </div>
         <div className={styles.infoLine}>
           <p>Value: </p>
           <p>
-            <Read amount={payload[0]?.payload.value} fixed={2} decimals={0} />{" "}
+            <Read amount={value} denom={denom} fixed={2} decimals={0} />{" "}
```

Comment:
Didn't we get rid of the need to pass decimals to the `Read` component?

Reply:
this for a currency amount, not a token denom. You can still override the decimals of the `read` component by passing. I experimented with a number of ways to simplify props passing here but this is the best and most flexible.